backend/infrastructure/api/endpoints/client.py

```diff
@@ -0,0 +1,71 @@
+from fastapi import APIRouter, HTTPException, Request
+from pydantic import BaseModel
+from infrastructure.mongo.client_handler import register_client, get_client
+
+class Client(BaseModel):
+    email: str
+    payment_address: str
+    delivery_address: str
+    nip: str
+    orders: str
+    password: str
+    company_name: str
+
+router = APIRouter()
+
+@router.post("/register")
+async def upload_client_endpoint(client_data: Client, request: Request):
+    if request.headers.get("Content-Type") != "application/json":
+        raise HTTPException(status_code=400, detail="Content-Type must be application/json")
```

Comment:
Please make some research about more global configuration of header validation, like we can suppose we support onyll application/json in whole app